dspace-api/src/test/java/org/dspace/content/EntityServiceImplTest.java

```diff
@@ -0,0 +1,224 @@
+        when(item.getID()).thenReturn(uuid);
+        when(entity.getRelationships()).thenReturn(relationshipList);
+        assertEquals("TestGetLeftRelations 0", relationshipList, entityService.getRightRelations(context, entity));
+    }
+
+    @Test
+    public void testGetRelationsByLabel() throws Exception {
+        Relationship relationship = mock(Relationship.class);
+        RelationshipType relationshipType = mock(RelationshipType.class);
+        relationship.setRelationshipType(relationshipType);
+        List<Relationship> relationshipList = new ArrayList<>();
+        relationshipList.add(relationship);
+        when(relationshipService.findAll(context)).thenReturn(relationshipList);
+        when(relationship.getRelationshipType()).thenReturn(relationshipType);
+        when(relationshipType.getLeftLabel()).thenReturn("leftLabel");
+        when(relationshipType.getRightLabel()).thenReturn("rightLabel");
+        assertEquals("TestGetRelationsByLabel 0", relationshipList,
+                entityService.getRelationsByLabel(context, "leftLabel"));
+    }
+
+    @Test
+    public void testGetAllRelationshipTypes() throws Exception {
+        List<MetadataValue> list = new ArrayList<>();
+        MetadataValue metadataValue = mock(MetadataValue.class);
+        list.add(metadataValue);
+        Item item = mock(Item.class);
+        RelationshipTypeDAO relationshipTypeDAO = mock(RelationshipTypeDAO.class);
+        Entity entity = mock(Entity.class);
+        Relationship relationship = mock(Relationship.class);
+        RelationshipType relationshipType = mock(RelationshipType.class);
+        relationship.setRelationshipType(relationshipType);
+        relationshipType.setLeftType(leftType);
+        relationshipType.setLeftType(rightType);
+        List<RelationshipType> relationshipTypeList = new ArrayList<>();
+        relationshipTypeList.add(relationshipType);
+        when(metadataValue.getValue()).thenReturn("testType");
+        when(entity.getItem()).thenReturn(item);
+        when(itemService.getMetadata(item, "relationship", "type", null, Item.ANY)).thenReturn(list);
+        when(relationshipTypeDAO.findAll(context, RelationshipType.class)).thenReturn(relationshipTypeList);
+        when(relationshipTypeService.findAll(context)).thenReturn(relationshipTypeList);
+        when(relationshipType.getLeftType()).thenReturn(leftType);
+        when(relationshipType.getRightType()).thenReturn(rightType);
+        when(entityTypeService.findByEntityType(context, "value")).thenReturn(leftType);
+        when(leftType.getID()).thenReturn(0);
+        when(rightType.getID()).thenReturn(1);
+        when(entityService.getType(context, entity)).thenReturn(leftType);
+        assertEquals("TestGetAllRelationshipTypes 0", relationshipTypeList,
+                entityService.getAllRelationshipTypes(context, entity));
+    }
+
+    @Test
+    public void testGetLeftRelationshipTypes() throws Exception {
```

Comment:
Again, here, could use some inline comments.  Any test method with more than ~10 lines should really have some human explanation...otherwise, developers are a bit stuck trying to parse what the code "means" and then determine whether the test is accurate based on what the code seems to be doing.